libraries/src/Router/ApiRouter.php

```diff
@@ -82,7 +56,15 @@ public function parseApiRoute($method = 'GET')
 	{
 		$method = strtoupper($method);
 
-		if (!array_key_exists($method, $this->routes))
+		$invalidMethod = true;
+
+		foreach ($this->routes as $route){
```

Comment:
I don't think this section works anymore. We just need to do this check after we've matched the route path.

Reply:
It works, I've tested the api. But if you think it is redundant to do the check at the beginning, I can throw the exception at the end, after we've checked if there is a matching route path.

Reply:
This looks like it's going to be problematic if you try to route when using the `PUT` method and the first route in the container doesn't support `PUT`.

```php
<?php

$router = new \Joomla\CMS\Router\ApiRouter;
$router->createCRUDRoutes(...$params);
$router->parseApiRoute('PUT');
```

Since the first route only supports `GET` it'll set the `$invalidMethod` flag to false right away.

Reply:
This works now with the break but having to loop through all the routes seems less optimal. We should probably just do an `in_array` check on `GET, POST, PUT, DELETE, HEAD, OPTIONS, TRACE, PATCH` as strings

Reply:
ok. I can change it for the next pull request.